sdks/java/io/redis/src/test/java/org/apache/beam/sdk/io/redis/RedisIOTest.java

```diff
@@ -0,0 +1,171 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.redis;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.ArrayList;
+
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.junit.Rule;
+import org.junit.Test;
+
+import redis.clients.jedis.Jedis;
+import redis.embedded.RedisServer;
+
+/**
+ * Test on the Redis IO.
+ */
+public class RedisIOTest {
+
+  @Rule public TestPipeline pipeline = TestPipeline.create();
+
+  @Test
+  public void testGetRead() throws Exception {
+    try (EmbeddedRedis embeddedRedis = new EmbeddedRedis()) {
+      ArrayList<KV<String, String>> expected = new ArrayList<>();
+      Jedis jedis = new Jedis("localhost", embeddedRedis.getPort());
+      for (int i = 0; i < 1000; i++) {
+        KV<String, String> kv = KV.of("Key " + i, "Value " + i);
+        jedis.set(kv.getKey(), kv.getValue());
+        expected.add(kv);
+      }
+      jedis.quit();
+
+      PCollection<KV<String, String>> readAll = pipeline.apply(
+          RedisIO.read()
+              .withConnectionConfiguration(RedisConnectionConfiguration.create()
+                  .withHost("localhost")
+                  .withPort(embeddedRedis.getPort())));
+
+      PAssert.thatSingleton(readAll.apply("Count", Count.<KV<String, String>>globally()))
+          .isEqualTo(1000L);
+      PAssert.that(readAll).containsInAnyOrder(expected);
+
+      pipeline.run();
+    }
+  }
+
+  @Test
+  public void testGetReadWithKeyPattern() throws Exception {
+    try (EmbeddedRedis embeddedRedis = new EmbeddedRedis()) {
+      Jedis jedis = new Jedis("localhost", embeddedRedis.getPort());
+      for (int i = 0; i < 1000; i++) {
+        KV<String, String> kv = KV.of("Key " + i, "Value " + i);
+        jedis.set(kv.getKey(), kv.getValue());
+      }
+      jedis.quit();
+
+      PCollection<KV<String, String>> read = pipeline.apply(
+          RedisIO.read()
+            .withConnectionConfiguration(RedisConnectionConfiguration.create()
+              .withHost("localhost")
+              .withPort(embeddedRedis.getPort()))
+            .withKeyPattern("Key 0"));
```

Comment:
Would be good if the test verified that we're matching a pattern rather than a single key - specify a pattern that matches multiple keys? And, for good measure, also have a test where the pattern matches no keys?